compiler/next/lib/parsing/chpl.ypp

```diff
@@ -1113,76 +1113,89 @@ use_stmt:
+                                     VisibilityClause::EXCEPT,
                                      $7);
   }
 | access_control TUSE expr TAS TUNDERSCORE TEXCEPT except_ls TSEMI
   {
     auto ident = toOwned(context->buildIdent(@5, $5));
     auto as = context->buildAsExpr(@3, @5, toOwned($3), std::move(ident));
-    auto locUseClause = context->makeSpannedLocation(@3, @7);
-    $$ = context->buildSingleUseStmt(@$, locUseClause, $1, toOwned(as),
-                                     UseClause::EXCEPT,
+    auto locVisibilityClause = context->makeSpannedLocation(@3, @7);
+    $$ = context->buildSingleUseStmt(@$, locVisibilityClause, $1, toOwned(as),
+                                     VisibilityClause::EXCEPT,
                                      $7);
   }
 | access_control TUSE expr TONLY opt_only_ls TSEMI
   {
-    auto locUseClause = context->makeSpannedLocation(@3, @5);
-    $$ = context->buildSingleUseStmt(@$, locUseClause, $1, toOwned($3),
-                                     UseClause::ONLY,
+    auto locVisibilityClause = context->makeSpannedLocation(@3, @5);
+    $$ = context->buildSingleUseStmt(@$, locVisibilityClause, $1, toOwned($3),
+                                     VisibilityClause::ONLY,
                                      $5);
   }
 | access_control TUSE expr TAS expr TONLY opt_only_ls TSEMI
   {
     auto as = context->buildAsExpr(@3, @5, toOwned($3), toOwned($5));
-    auto locUseClause = context->makeSpannedLocation(@3, @7);
-    $$ = context->buildSingleUseStmt(@$, locUseClause, $1, toOwned(as),
-                                     UseClause::ONLY,
+    auto locVisibilityClause = context->makeSpannedLocation(@3, @7);
+    $$ = context->buildSingleUseStmt(@$, locVisibilityClause, $1, toOwned(as),
+                                     VisibilityClause::ONLY,
                                      $7);
   }
 | access_control TUSE expr TAS TUNDERSCORE TONLY opt_only_ls TSEMI
   {
     auto ident = toOwned(context->buildIdent(@5, $5));
     auto as = context->buildAsExpr(@3, @5, toOwned($3), std::move(ident));
-    auto locUseClause = context->makeSpannedLocation(@3, @7);
-    $$ = context->buildSingleUseStmt(@$, locUseClause, $1, toOwned(as),
-                                     UseClause::ONLY,
+    auto locVisibilityClause = context->makeSpannedLocation(@3, @7);
+    $$ = context->buildSingleUseStmt(@$, locVisibilityClause, $1, toOwned(as),
+                                     VisibilityClause::ONLY,
                                      $7);
   }
 ;
 
 import_stmt:
   access_control TIMPORT import_ls TSEMI
-    {
-      $$ = TODOSTMT(@$);
-      context->visibility = Decl::DEFAULT_VISIBILITY;
-    }
+  {
+    auto comments = context->gatherComments(@1);
+    auto node = Import::build(BUILDER, LOC(@$), $1, context->consumeList($3));
+    CommentsAndStmt cs = { .comments=comments, .stmt=node.release() };
+    $$ = context->finishStmt(cs);
+  }
 ;
 
 import_expr:
   expr
```

Comment:
Since this can parse any expr at all, we'll need to check in the ParserContext helper I'm hoping you'll add to make an Import - we'll need to check that the expr is one of the supported ones and not, say, a function call. This applies also to the Use statement. We should be able to get a parser error out if it's a function call (say). Let's add a test case for use and import for these.

Reply:
I wasn't planning on making an `Import` since the nodes are relatively simple to build, but it sounds like I should make one now...

Reply:
At the very least let's get a test case for things like `import 1+1` and make sure we get an error out.